lib/autoscale/aws/autoscaler.go

```diff
@@ -160,6 +160,27 @@ func (a *Autoscaler) DescribeInstance(ctx context.Context, instanceID string) (*
 	return resp.Reservations[0].Instances[0], nil
 }
 
+// DescribeInstancesWithSourceDestinationCheck returns all instances from the
+// specified list that have source/destination check enabled.
+func (a *Autoscaler) DescribeInstancesWithSourceDestinationCheck(ctx context.Context, instanceIDs []string) (result []*ec2.Instance, err error) {
+	resp, err := a.Cloud.DescribeInstancesWithContext(ctx, &ec2.DescribeInstancesInput{
+		InstanceIds: aws.StringSlice(instanceIDs),
```

Comment:
Not so much on the response side, but on the request side I'm curious how AWS responds when passed 1k instances?

I don't see any specific notes in the AWS API docs, except when not using instance id's and instead using maxResults, the range can be 5-1000 (Source: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstances.html). 

It might be worth considering splitting the list into separate calls in groups of a couple hundred just incase?